tests/phpunit/tests/dependencies/scripts.php

```diff
@@ -63,6 +63,67 @@ public function test_wp_enqueue_script() {
 		$this->assertSame( '', get_echo( 'wp_print_scripts' ) );
 	}
 
+	/**
+	 * Test old and new in_footer logic.
+	 */
+	public function test_old_and_new_in_footer_scripts() {
+		wp_register_script( 'header-old', '/header-old.js', array(), null, false );                                         // In head.
```

Comment:
I'd suggest moving these inline comments to the line above the code, for readability.